registrations/migrations/0024_signup_and_signup_group_protected_data.py

```diff
@@ -0,0 +1,156 @@
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                (
+                    "extra_info",
+                    encrypted_fields.fields.EncryptedTextField(
+                        blank=True, default=None, null=True, verbose_name="Extra info"
+                    ),
+                ),
+                (
+                    "registration",
+                    models.ForeignKey(
+                        on_delete=deletion.PROTECT,
+                        related_name="signupprotecteddatas",
+                        to="registrations.registration",
+                    ),
+                ),
+                (
+                    "date_of_birth",
+                    encrypted_fields.fields.EncryptedDateField(
+                        blank=True, null=True, verbose_name="Date of birth"
+                    ),
+                ),
+                (
+                    "signup",
+                    models.OneToOneField(
+                        default=None,
+                        null=True,
+                        on_delete=deletion.SET_NULL,
+                        related_name="protected_data",
+                        to="registrations.signup",
+                    ),
+                ),
+            ],
+            options={
+                "abstract": False,
+            },
+        ),
+        migrations.CreateModel(
+            name="SignUpGroupProtectedData",
+            fields=[
+                (
+                    "id",
+                    models.AutoField(
+                        auto_created=True,
+                        primary_key=True,
+                        serialize=False,
+                        verbose_name="ID",
+                    ),
+                ),
+                (
+                    "extra_info",
+                    encrypted_fields.fields.EncryptedTextField(
+                        blank=True, default=None, null=True, verbose_name="Extra info"
+                    ),
+                ),
+                (
+                    "registration",
+                    models.ForeignKey(
+                        on_delete=deletion.PROTECT,
+                        related_name="signupgroupprotecteddatas",
+                        to="registrations.registration",
+                    ),
+                ),
+                (
+                    "signup_group",
+                    models.OneToOneField(
+                        default=None,
+                        null=True,
+                        on_delete=deletion.SET_NULL,
+                        related_name="protected_data",
+                        to="registrations.signupgroup",
+                    ),
+                ),
+            ],
+            options={
+                "abstract": False,
+            },
+        ),
+        migrations.RunPython(forwards_func, migrations.RunPython.noop),
```

Comment:
I don't know is this big deal, but we are now losing data or `extra_info` and `date_of_birth` fields when migrating back to `0023_rename_timestamp_fields`. If the registration would run in the production it could be.

Reply:
This has now been improved by introducing a `backwards_func` data migration.